### ISSUE TITLE
Add a test of the 'git describe' tagging behavior

### ISSUE THREAD

**Issue by lavalamp** [kubernetes/kubernetes#27368]
We accidentally undid part of the git tag that the release script produces. We need a test (hack/verify-git-tag.sh?) that verifies that the tag is correct so we catch this immediately.

**Comment by ixdy** [comment]
what does "tag is correct" mean?

**Comment by david-mcmahon** [comment]
Yes, I understand the sense of urgency here, but I'm having trouble grok'ing this myself. 
I'm not exactly sure what it is we want to check here.

**Comment by david-mcmahon** [comment]
@lavalamp can you elaborate further here on what exactly we want to check or can we just close this?

**Comment by lavalamp** [comment]
If the tag produced by git describe is wrong, then our post submit build job and post submit test jobs silently stop functioning correctly. That is, the post submit test jobs start looking for a build in the wrong place and pick up the last build pushed there. This caused our test jobs to be useless for a couple days until we noticed the problem.

**Comment by lavalamp** [comment]
The request here is for a test that will notice this condition and fail visibly.

**Comment by ixdy** [comment]
I'm still not sure what "tag is wrong" means. Do you have an example of where this failed?

**Comment by dims** [comment]
@lavalamp @ixdy - there's at least one regex (https://github.com/kubernetes/kubernetes/blob/5e29e1ee05556800cd2fe0dc5341d5de54b4ab47/hack/lib/version.sh#L78) we could strengthen that. (or close this issue!)
